test/ecto_test.exs

```diff
@@ -109,4 +109,90 @@ defmodule EctoTest do
 
     query
   end
+
+  defmodule Post do
+    use Ecto.Schema
+
+    schema "posts" do
+      field :title, :string, default: "default title"
+      belongs_to :author, Author
+      has_one :permalink, Permalink
+      has_many :comments, Comment
+      many_to_many :tags, Tag, join_through: "posts_tags"
+    end
+  end
+
+  defmodule Author do
+    use Ecto.Schema
+
+    schema "authors" do
+      field :name, :string
+      has_many :posts, Post
+    end
+  end
+
+  defmodule Permalink do
+    use Ecto.Schema
+
+    schema "permalinks" do
+      field :url, :string
+      belongs_to :post, Post
+    end
+  end
+
+  defmodule Comment do
+    use Ecto.Schema
+
+    schema "comments" do
+      field :url, :string
+    end
+  end
+
+  defmodule Tag do
+    use Ecto.Schema
+
+    schema "tags" do
+      field :name, :string
+      many_to_many :posts, Post, join_through: "posts_tags"
+    end
+  end
+
+  test "Ecto.reset_fields/2 resets fields" do
+    post = %Post{
+      title: "hello world",
+      author: %Author{},
+      permalink: %Permalink{},
+      comments: [
+        %Comment{},
+        %Comment{}
+      ],
+      tags: [
+        %Tag{},
+        %Tag{},
+        %Tag{}
+      ]
+    }
```

Comment:
```suggestion
      comments: [%Comment{}, %Comment{}],
      tags: [%Tag{}, %Tag{}, %Tag{}]
    }
```
sorry for the nit. it's just easier to grep the test file when there are less lines